src/groups/mqb/mqbblp/mqbblp_localqueue.cpp

```diff
@@ -515,7 +516,12 @@ void LocalQueue::postMessage(const bmqp::PutHeader&              putHeader,
                                    appData->length());
 
         if (haveReceipt && refCount) {
-            d_hasNewMessages = true;
+            if (!d_hasNewMessages) {
+                BSLS_ASSERT_SAFE(d_notifications_p);
+
+                d_hasNewMessages = true;
+                d_notifications_p->push_back(d_state_p->queue());
```

Comment:
We add each queue to the notification list only once, when we switch the flag from false to true